sdk/communication/azure-communication-networktraversal/README.md

```diff
@@ -0,0 +1,134 @@
+    .credential(new DefaultAzureCredentialBuilder().build())
+    .buildClient();
+```
+
+### AzureKeyCredential Authentication
+Network Traversal uses HMAC authentication with the resource access key.
+The access key can be used to create an AzureKeyCredential and provided to the `CommunicationRelayClientBuilder` via the credential() function. Endpoint and httpClient must also be set via the endpoint() and httpClient() functions respectively.
+
+<!-- embedme ./src/samples/java/com/azure/communication/networktraversal/ReadmeSamples.java#L21-L28 -->
+```java
+// You can find your endpoint and access key from your resource in the Azure Portal
+String endpoint = "https://<RESOURCE_NAME>.communication.azure.com";
+AzureKeyCredential keyCredential = new AzureKeyCredential("<access-key>");
+
+CommunicationRelayClient communicationRelayClient = new CommunicationRelayClientBuilder()
+    .endpoint(endpoint)
+    .credential(keyCredential)
+    .buildClient();
+```
+
+### Connection String Authentication
+Alternatively, you can provide the entire connection string using the connectionString() function instead of providing the endpoint and access key.
+<!-- embedme ./src/samples/java/com/azure/communication/networktraversal/ReadmeSamples.java#L39-L44 -->
+```java
+// You can find your connection string from your resource in the Azure Portal
+String connectionString = "<connection_string>";
+
+CommunicationRelayClient communicationRelayClient = new CommunicationRelayClientBuilder()
+    .connectionString(connectionString)
+    .buildClient();
+```
+
+## Key concepts
+
+`CommunicationRelayClient` and `CommunicationRelayAsyncClient` provide the functionalities to manage users and user tokens.
+
+## Examples
+
+### Getting a new Relay Configuration
+Use the `createUser` function to create a new user from CommunicationIdentityClient
+Use the `getRelayConfiguration` function to get a Relay Configuration
+
+<!-- embedme ./src/samples/java/com/azure/communication/networktraversal/ReadmeSamples.java#L73-L74 -->
+```java
+CommunicationIdentityClient communicationIdentityClient = new CommunicationIdentityClientBuilder()
+            .connectionString(connectionString)
+            .buildClient();
+
+CommunicationUserIdentifier user = communicationIdentityClient.createUser();
+System.out.println("User id: " + user.getId());
+
+CommunicationRelayConfiguration config = communicationRelayClient.getRelayConfiguration(user);
+```
+
+## Troubleshooting
+
+All user token service operations will throw an exception on failure.
+<!-- embedme ./src/samples/java/com/azure/communication/networktraversal/ReadmeSamples.java#L139-L143 -->
+```java
+try {
+    CommunicationUserIdentifier user = communicationIdentityClient.createUser();
+    CommunicationRelayClient communicationRelayClient = createCommunicationNetworkTraversalClient();
+    CommunicationRelayConfiguration config = communicationRelayClient.getRelayConfiguration(user);
+} catch (RuntimeException ex) {
```

Comment:
Does the client throw a more specific exception here? Ideally, the client is expected to throw a specific exception type that is derived from `AzureException`.

Reply:
It throws a CommunicationErrorResponseException, this comes from the auto generated code.

Reply:
Then the sample should include the above exception instead of catching RuntimeException as it's too broad.

Reply:
Also, this doesn't really help with troubleshooting. Since this is under the `Troubleshooting` section, you should also include what to do to fix the issue or how to investigate when an exception is thrown.

Reply:
Changed the exception. About the troubleshooting that will be added later with links to the documentation with the steps and proper links.